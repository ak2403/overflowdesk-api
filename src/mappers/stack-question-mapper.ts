import { StackQuestion } from "../types/mappers/stack-question";
import { StackOverflowQuestion } from "../types/models/stackoverflow";

type PrimaryQuestionProps = Omit<
  StackOverflowQuestion,
  "owner" | "last_activity_date" | "content_license"
>;

export class StackQuestionMapper {
  static transform(question: PrimaryQuestionProps): StackQuestion {
    const { creation_date, question_id, link, score, tags, title, view_count } =
      question;

    return {
      createdDate: creation_date,
      id: question_id,
      link,
      score,
      tags,
      title,
      viewCount: view_count,
    };
  }
}
