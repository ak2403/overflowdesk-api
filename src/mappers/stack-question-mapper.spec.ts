import { StackQuestionMapper } from "./stack-question-mapper";

describe("StackQuestionMapper", () => {
  it("transform question response to valid model", () => {
    const result = StackQuestionMapper.transform({
      tags: ["tag"],
      is_answered: true,
      view_count: 24,
      answer_count: 1,
      score: 1,
      creation_date: 1547025378,
      question_id: 54106638,
      link: "demo-link",
      title: "demo-title",
    });

    expect(result).toMatchSnapshot();
  });
});
