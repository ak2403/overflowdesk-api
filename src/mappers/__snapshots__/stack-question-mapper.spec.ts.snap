// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackQuestionMapper runs 1`] = `
{
  "createdDate": 1547025378,
  "id": 54106638,
  "link": "demo-link",
  "score": 1,
  "tags": [
    "tag",
  ],
  "title": "demo-title",
  "viewCount": 24,
}
`;
